src/odemis/driver/test/hamamatsurx_test.py

```diff
@@ -0,0 +1,704 @@
+        f.result()  # wait for the position to be set
+        f = self.spectrograph.moveAbs({"grating": pos_grating})
+        f.result()  # wait for the position to be set
+        f = self.spectrograph.moveAbs({"slit-in": pos_slit})
+        f.result()  # wait for the position to be set
+
+        # VAs should have same values as HW positions
+        self.assertAlmostEqual(self.spectrograph.position.value["wavelength"], pos_wl)
+        self.assertEqual(self.spectrograph.position.value["grating"], pos_grating)
+        self.assertAlmostEqual(self.spectrograph.position.value["slit-in"], pos_slit)
+
+
+# Inheritance order is important for setUp, tearDown
+#@skip("simple")
+class TestHamamatsurxCamGenericCam(VirtualTestCam, unittest.TestCase):
+    """
+    Test directly the Hamamatsu streak camera class.
+    Run the generic camera test cases.
+    """
+    camera_type = None
+    camera_kwargs = None
+
+    @classmethod
+    def setUpClass(cls):
+
+        cls.streakcam = hamamatsurx.StreakCamera("streak cam", "streakcam", host="DESKTOP-E6H9DJ0", port=1001,
+                                                    children=children)
+
+        for child in cls.streakcam.children.value:
+            if child.name == CONFIG_READOUTCAM["name"]:
+                cls.camera = child
+            if child.name == CONFIG_STREAKUNIT["name"]:
+                cls.streakunit = child
+            if child.name == CONFIG_DELAYBOX["name"]:
+                cls.delaybox = child
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.streakcam.terminate()
+
+
+CONFIG_SED = {"name": "sed", "role": "sed", "channel":5, "limits": [-3, 3]}
+CONFIG_SCANNER = {"name": "scanner", "role": "ebeam", "limits": [[0, 5], [0, 5]],
+                  "channels": [0, 1], "settle_time": 10e-6, "hfw_nomag": 10e-3}
+CONFIG_SEM = {"name": "sem", "role": "sem", "device": "/dev/comedi0",
+              "children": {"detector0": CONFIG_SED, "scanner": CONFIG_SCANNER}
+              }
+
+
+#@skip("simple")
+class TestHamamatsurxCamGenericCamSynchronized(VirtualTestSynchronized, unittest.TestCase):
+    """
+    Test the synchronizedOn(Event) interface with real streak camera HW, using the fake SEM.
+    Run the generic camera test cases.
+    """
+    camera_type = None
+    camera_kwargs = None
+
+    @classmethod
+    def setUpClass(cls):
+
+        cls.streakcam = hamamatsurx.StreakCamera("streak cam", "streakcam", host="DESKTOP-E6H9DJ0", port=1001,
+                                                    children=children)
+
+        for child in cls.streakcam.children.value:
+            if child.name == CONFIG_READOUTCAM["name"]:
+                cls.ccd = child
+            if child.name == CONFIG_STREAKUNIT["name"]:
+                cls.streakunit = child
+            if child.name == CONFIG_DELAYBOX["name"]:
+                cls.delaybox = child
+
+        # TODO set up SEM scanner, can we use the setUpClass from generic test?
```

Comment:
Indeed, if you set `.camera_type` to `hamamatsurx.StreakCamera`, and `camera_kwargs` to KWARGS, you should be able to just call the `super()`.  And probably you can drop the children related code, as they don't seem used anyway.
So you might be able to entirely drop `setUpClass` and `tearDownClass`.